contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -91,20 +113,11 @@ static void handle_trilogy_error(struct trilogy_ctx *ctx, int rc, const char *ms
     }
 }
 
-static void free_trilogy(struct trilogy_ctx *ctx)
-{
-    if (ctx->conn.socket != NULL) {
-        trilogy_free(&ctx->conn);
-    }
-}
-
 static VALUE allocate_trilogy(VALUE klass)
 {
     struct trilogy_ctx *ctx;
 
-    VALUE obj = Data_Make_Struct(klass, struct trilogy_ctx, NULL, free_trilogy, ctx);
-
-    memset(ctx->server_version, 0, sizeof(ctx->server_version));
```

Comment:
I don't think this `memset` was ever necessary since `Make_Struct` use `ZALLOC`.